manifests/0000_30_cluster-api_09_admission-policies.yaml

```
@@ -239,6 +239,51 @@ data:
     ---
     apiVersion: admissionregistration.k8s.io/v1
     kind: ValidatingAdmissionPolicy
+    metadata:
+      name: openshift-prevent-mapi-machineset-create-when-capi-exists
+    spec:
+      failurePolicy: Fail
+      paramKind:
+        apiVersion: cluster.x-k8s.io/v1beta1
+        kind: MachineSet
+      matchConstraints:
+        resourceRules:
+          - apiGroups: ["machine.openshift.io"]
+            apiVersions: ["*"]
+            operations: ["CREATE"]
+            resources: ["machinesets"]
+      # Requests must satisfy every matchCondition to reach the validations
+      matchConditions:
+        - name: check-param-match
+          expression: 'object.metadata.name == params.metadata.name'
+      # All validations must evaluate to true
+      validations:
+      - expression: 'object.spec.authoritativeAPI == "ClusterAPI"'
+        messageExpression: "'Can\\'t create Machine API MachineSet ' + object.metadata.name + ' with authoritativeAPI=' + object.spec.?authoritativeAPI.orValue('<unset>') + ' because a Cluster API MachineSet with the same name already exists.'"
```

We could add an extra comment.
```suggestion
      # only allow creation of  Machine API MachineSet if spec.authoritativeAPI == ClusterAPI
      - expression: 'object.spec.authoritativeAPI == "ClusterAPI"'
        messageExpression: "'Can\\'t create Machine API MachineSet ' + object.metadata.name + ' with authoritativeAPI=' + object.spec.?authoritativeAPI.orValue('<unset>') + ' because a Cluster API MachineSet with the same name already exists.'"
```